src/test/app/forms/models/defendantDetails.ts

```diff
@@ -0,0 +1,205 @@
+
+    it('should accept defendant details for individual type when individual details are null', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.INDIVIDUAL, '', '', undefined, undefined))
+
+        expect(errors.length).to.equal(1)
+        expectValidationError(errors, ValidationErrors.FULLNAME_REQUIRED)
+      })
+    })
+
+    it('should accept defendant details for organisation type when organisation details are valid', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.ORGANISATION, undefined, undefined, 'organisation name', '12345678'))
+
+        expect(errors.length).to.equal(0)
+      })
+    })
+
+    it('should reject defendant details for organisation type when organisation name is null', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.ORGANISATION, undefined, undefined, null, '12345678'))
+
+        expect(errors.length).to.equal(1)
+        expectValidationError(errors, ValidationErrors.ORGANISATION_NAME_REQUIRED)
+      })
+    })
+
+    it('should reject defendant details for organisation type when organisation name is undefined', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.ORGANISATION, undefined, undefined, undefined, '12345678'))
+
+        expect(errors.length).to.equal(1)
+        expectValidationError(errors, ValidationErrors.ORGANISATION_NAME_REQUIRED)
+      })
+    })
+
+    it('should reject defendant details for organisation type when organisation name length is more than 255', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.ORGANISATION,
+          undefined, undefined, randomstring.generate( 256 ), '12345678'))
+
+        expect(errors.length).to.equal(1)
+        expectValidationError(errors, ValidationErrors.CONTENT_TOO_LONG)
+      })
+    })
+
+    it('should reject defendant details for organisation type when companyHouseNumber length is more than 8', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.ORGANISATION,
+          undefined, undefined, 'organisation', randomstring.generate( 9 )))
+
+        expect(errors.length).to.equal(1)
+        expectValidationError(errors, ValidationErrors.CONTENT_TOO_LONG)
+      })
+    })
+
+    it('should reject defendant details for individual type when full name length is more than 70', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.INDIVIDUAL,
+          undefined, randomstring.generate( 71 ), undefined, undefined))
+
+        expect(errors.length).to.equal(2)
```

Comment:
shouldn't we be checking the one error here. And if this test case generates two errors, description should mention that.